### PR TITLE
[DRAFT/Conversation] gpg: raise a clearer error when a key to install is invalid

### PR DESCRIPTION
**Pull request by lengau**
- [x] Have you followed the guidelines for contributing?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `tox`?

-----

### CODE REVIEW

File: craft_archives/repo/apt_key_manager.py [hunk 1]
```diff
@@ -161,7 +163,10 @@ def install_key(self, *, key: str) -> None:
         :raises: AptGPGKeyInstallError if unable to install key.
         """
         logger.debug(f"Importing key {key}")
-        fingerprints = self.get_key_fingerprints(key=key)
+        try:
+            fingerprints = self.get_key_fingerprints(key=key)
+        except subprocess.CalledProcessError as error:
+            raise errors.AptGPGKeyInstallError(error.output.decode(), key=key)
```
Comment (tigarmo) [inline craft_archives/repo/apt_key_manager.py:169]: I think we need to look more carefully at this. I was experimenting with this yesterday and I think gpg writes its error messages to `stderr`, so this output is always empty which is annoying. We should make use of the integrated tests to assess that we're correctly capturing gpg's messages.

At the same time, there's this whole craft-cli design philosophy where the actual noisy error messages should *not* show up on the "default" output that the user sees when running snapcraft or rockcraft or what have you; so we probably need to rethink these error classes to have a better split between the 'brief' and 'details' messages. So I suggest we put a pin on this for now.
What do you think?

**Review by tigarmo — CHANGES_REQUESTED** [review]

File: craft_archives/repo/apt_key_manager.py (hunk 1 above)
Comment (lengau) [reply craft_archives/repo/apt_key_manager.py:169]: Agreed. This one's non-urgent, but popped up when I was doing Rockcraft

Comment (lengau) [reply craft_archives/repo/apt_key_manager.py:169]: (Converting to draft so we can continue conversation without this looming)

**Comment by lengau** [comment]
No longer relevant

**Pull request closed without merging**